### PR TITLE
CSVS-35 User creation event processing was changed

### PR DESCRIPTION
**Pull request by MedvedevBW**

### CODE REVIEW

File: docs/logic.md [hunk 1]
```diff
@@ -22,14 +22,15 @@ The processing of VM creation event consists of the following steps: <br />
 
 The processing of account creation event consists of the following steps: <br />
   * a) get the account users <br />
-  * b) get a tag list of each user and select only token tags <br />
-  * с) if the list of tokens is empty or does not contain both account tokens ('RO' token and 'RW' token): <br />
+  * b) get a tag list of each user and select only vault tags <br />
+  * с) if the list of tokens is empty or does not contain both vault tokens ('RO' token, 'RW' token, 'vaultPrefix', 'vaultUrl'): <br />
```
Comment (placccebo) [inline docs/logic.md]: if the list of tokens is empty or does not contain the mandatory vault tokens

File: docs/logic.md [hunk 2]
```diff
@@ -22,14 +22,15 @@ The processing of VM creation event consists of the following steps: <br />
...
     * 1) try to retrieve account tokens from zookeeper <br />
     * 2) if the tokens don't exist: <br />
       * a) the tokens are created in vault <br />
       * b) zookeeper nodes are created for keeping account tokens <br />
       * c) write the vault tokens to zookeeper (in case of an exception, the tokens will be revoked) <br />
-    * 3) write the tokens in CloudStack as user tags for each user <br />
+    * 3) create 'vaultPrefix' or 'vaultUrl' tags if it does not exist <br />
```
Comment (placccebo) [inline docs/logic.md]: create 'vaultPrefix' and/or 'vaultUrl' tags if they do not exist

File: docs/logic.md [hunk 3]
```diff
@@ -22,14 +22,15 @@ The processing of VM creation event consists of the following steps: <br />
...
+    * 4) write the tag in CloudStack as user tags for users which do not have all tags for vault data  <br />
```
Comment (placccebo) [inline docs/logic.md]: write the missing tags in CloudStack as user tags for each user

File: docs/logic.md [hunk 4]
```diff
@@ -41,7 +42,8 @@ The processing of user creation event consists of the following steps: <br />
       * a) the tokens are created in vault <br />
       * b) zookeeper nodes are created for keeping account tokens <br />
       * c) write the vault tokens to zookeeper (in case of an exception, the tokens will be revoked) <br />
-  * e) write the tokens in CloudStack as user tags <br />
+    * 3) create 'vaultPrefix' or 'vaultUrl' tags if it does not exist <br />
```
Comment (placccebo) [inline docs/logic.md]: see above

**Review by placccebo — CHANGES_REQUESTED** [review]

File: src/main/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultController.scala [hunk 5]
```diff
@@ -82,8 +82,31 @@ class CloudStackVaultController(vaultService: VaultService,
...
+    val newVaultKeyspaceTagKeyList = Set(Tag.Key.VaultHost, Tag.Key.VaultPrefix).collect {
+      case tagKey if !currentVaultTags.exists(_.key == tagKey) =>
+        tagKey match {
+          case Tag.Key.VaultHost => Tag.createTag(Tag.Key.VaultHost, vaultApiPath)
+          case Tag.Key.VaultPrefix => Tag.createTag(Tag.Key.VaultPrefix, getAccountEntitySecretPath(accountId))
+        }
+    }
 
-    handleAccountCreate(accountId)
+    val completeVaultTags: Set[Tag] = currentVaultTags ++ newVaultTokenTagKeyList ++ newVaultKeyspaceTagKeyList
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultController.scala]: convert "completeVaultTags" to List here  (before if)

File: src/main/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultController.scala [hunk 6]
```diff
@@ -96,11 +119,28 @@ class CloudStackVaultController(vaultService: VaultService,
   def handleAccountCreate(accountId: UUID): Unit = {
     logger.debug(s"handleAccountCreate(accountId: $accountId)")
 
-    val usersWithUnrecordedVaultTags = getAccountUsersWithUnrecordedTokenTags(accountId)
+    val usersIds = cloudStackService.getUserIdsByAccountId(accountId)
+
+    val currentVaultTags = getCurrentUsersVaultTags(usersIds)
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultController.scala]: getCurrentVaultTagsOfUsers

**Review by placccebo — CHANGES_REQUESTED** [review]

File: docs/logic.md (hunk 2 above)
Comment (MedvedevBW) [reply docs/logic.md]: not actual

File: docs/logic.md (hunk 3 above)
Comment (MedvedevBW) [reply docs/logic.md]: not actual

File: docs/logic.md (hunk 1 above)
Comment (MedvedevBW) [reply docs/logic.md]: ok

File: docs/logic.md (hunk 4 above)
Comment (MedvedevBW) [reply docs/logic.md]: ok

File: src/main/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultController.scala (hunk 5 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultController.scala]: ok

File: src/main/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultController.scala (hunk 6 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/vault/server/controllers/CloudStackVaultController.scala]: ok

**Review by placccebo — APPROVED** [review]

**Pull request merged**